components/docked-composer/index.jsx

```diff
@@ -0,0 +1,168 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Docked Composer design pattern](https://lightningdesignsystem.com/components/docked-composer/) in React.
+// Based on SLDS v2.9.4
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import shortid from 'shortid';
+
+import { DOCKED_COMPOSER } from '../../utilities/constants';
+
+import Button from '../button';
+
+const propTypes = {
+	actionButton: PropTypes.node,
+	composerBody: PropTypes.node,
+	header: PropTypes.shape({
+		icon: PropTypes.node,
+		label: PropTypes.string,
```

Comment:
`labels` should  be top-level props. This is because all labels are typically stored in a text file somewhere in a database by language. It always makes them easier to find.

`labels.heading` is a good choice